### PR TITLE
CIDM-455 Add CORS filter support

### PR DESCRIPTION
**Pull request by teriyakichild**

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/rackerlabs/puppet-repose/pull/53" target="_blank">ReviewNinja

**Comment by teriyakichild** [comment]
@bjoshua  I got the tests to pass!!!

File: manifests/filter/cors.pp [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+# https://www.somehost.com:443) 
+# Defaults to <tt>{ 10 => { 'is_regex' => 'true', 'origin' => '.*' }</tt>
+#
+# [*allowed_methods*]
+# Array of HTTP VERBS. Allows you to specify which HTTP methods are allowed
+# for all resources (when specified under <cross-origin-resource-sharing>).
+# Defaults to <tt>undef</tt>
+#
+# [*resources*]
+# Hash of resources, name of hash being the resource path, hash cintains allowed_methods (Array of HTTP verbs) and comment (XML/HTTP style comment). Specifies
```
Comment (teriyakichild) [inline manifests/filter/cors.pp]: @bjoshua The documentation specifies that the "resources" parameter should be a hash but the template expects something like this:
```
[
  {
    'name': '/v2.0/tokens',
    'allowed_methods': [
      'POST',
      'GET',
      'OPTIONS',
    ]
  }
]
```

I expected something more like this:

```
{
  '/v2.0/tokens': [
    'POST',
    'GET',
    'OPTIONS',
  ]
}
```

**Comment by bjoshua** [comment]
LGTM

**Pull request merged**